koku/reporting/migrations/0154_gcp_partitioned.py

```diff
@@ -0,0 +1,92 @@
+# Generated by Django 3.1.3 on 2020-12-02 17:45
+from django.db import migrations
+
+from koku import pg_partition as ppart
+
+
+def convert_to_partitioned(table_name):
+    # Resolve the current schema name
+    target_schema = ppart.resolve_schema(ppart.CURRENT_SCHEMA)
+    # This is the table we will model from
+    source_table = table_name
+    # This is the target table's name (it will be renamed during the conversion to the source table name)
+    target_table = f"p_{source_table}"
+
+    # We'll want a new sequence copied from the original sequence
+    new_seq = ppart.SequenceDefinition(
+        target_schema,
+        f"{target_table}_id_seq",
+        copy_sequence={"schema_name": target_schema, "table_name": source_table, "column_name": "id"},
+    )
+    # We want to change the target tables's 'id' column default
+    target_identity_col = ppart.ColumnDefinition(target_schema, target_table, "id", default=ppart.Default(new_seq))
+    # We also need to include the identity col as part of the primary key definition
+    new_pk = ppart.PKDefinition(f"{target_table}_pkey", ["usage_start", "id"])
+
+    # Init the converter
+    p_converter = ppart.ConvertToPartition(
+        source_table,
+        "usage_start",
+        target_table_name=target_table,
+        partition_type=ppart.PARTITION_RANGE,
+        pk_def=new_pk,
+        col_def=[target_identity_col],
+        target_schema=target_schema,
+        source_schema=target_schema,
+    )
+
+    # Push the button, Frank.
+    p_converter.convert_to_partition()
+
+
+def convert_gcpcostentrylineitem_to_partitioned(apps, schema_editor):
+    convert_to_partitioned("reporting_gcpcostentrylineitem")
```

Comment:
Let's just stick with line item daily summary for now. I don't think we'll be using these long enough to worry about them being partitioned as well.

Reply:
No prob. I thought it would be an interesting experiment since it was to be short-lived anyway.

Reply:
Cody will make the changes as he's under the hood now.